### PR TITLE
windows: added skip_taskbar

### LINKED ISSUE #4341: Add way to skip adding window to taskbar
There should be a property similar to `no-frame` named `skip-taskbar` (or something similar) that doesn't add the window to the taskbar when set to `true`.

For now, I am modifying the winit `Window` directly.
```rust
win.window().with_winit_window(|winit_win| {
	winit_win.set_skip_taskbar(true);
});
```

### PR DESCRIPTION
**Pull request by ghost**
Fixes #4341

### CODE REVIEW

File: docs/src/content/docs/reference/window/window.mdx [hunk 1]
```diff
@@ -48,6 +48,11 @@ The window icon shown in the title bar or the task bar on window managers suppor
 Whether the window should be borderless/frameless or not.
 </SlintProperty>
 
+### skip-taskbar
+<SlintProperty propName="no-frame" typeName="bool" defaultValue="false">
```
Comment (tronical) [inline docs/src/content/docs/reference/window/window.mdx:52]: This looks like the wrong property name?

File: internal/backends/qt/qt_window.rs [hunk 2]
```diff
@@ -1943,14 +1944,15 @@ impl WindowAdapter for QtWindow {
...
-                      fullscreen as "bool", minimized as "bool", maximized as "bool"] {
+                      fullscreen as "bool", minimized as "bool", maximized as "bool", skip_taskbar as "bool"] {
 
             if (size != widget_ptr->size()) {
                 widget_ptr->resize(size.expandedTo({1, 1}));
             }
 
             widget_ptr->setWindowFlag(Qt::FramelessWindowHint, no_frame);
             widget_ptr->setWindowFlag(Qt::WindowStaysOnTopHint, always_on_top);
+            widget_ptr->setWindowFlag(Qt::Tool, skip_taskbar);
```
Comment (tronical) [inline internal/backends/qt/qt_window.rs:1955]: Have you tried this out? For me the effect on macOS seems... less than desirable.

File: internal/backends/winit/winitwindowadapter.rs [hunk 3]
```diff
@@ -13,6 +13,8 @@ use std::rc::Weak;
 #[cfg(target_arch = "wasm32")]
 use winit::platform::web::WindowExtWebSys;
 #[cfg(target_family = "windows")]
+use winit::platform::windows::WindowAttributesExtWindows;
+#[cfg(target_family = "windows")]
```
Comment (tronical) [inline internal/backends/winit/winitwindowadapter.rs:17]: If this is supported only on Windows, shouldn't this limitation be documented?

File: internal/backends/qt/qt_window.rs (hunk 2 above)
Comment (tronical) [reply internal/backends/qt/qt_window.rs:1955]: For me setting `Qt::Tool` to true on a window makes it stay on top of _all other windows_ in the application. So with a multi-window application that seems not what we want. (this is on macOS)

Comment (ogoffart) [reply internal/backends/qt/qt_window.rs:1955]: i'm not sure about that. Maybe Qt::Dialog?

File: docs/src/content/docs/reference/window/window.mdx [hunk 4]
```diff
@@ -48,6 +48,11 @@ The window icon shown in the title bar or the task bar on window managers suppor
 Whether the window should be borderless/frameless or not.
 </SlintProperty>
 
+### skip-taskbar
+<SlintProperty propName="skip-taskbar" typeName="bool" defaultValue="false">
+Whether to show or hide the window icon in the taskbar. (only supported on `Windows`)
```
Comment (tronical) [inline docs/src/content/docs/reference/window/window.mdx:53]: I can't find any other use of Windows in backticks in the docs, so proposing an alternative:

```suggestion
Whether to show or hide the window icon in the taskbar. This is feature only supported on Windows.
```

**Review by tronical — APPROVED** [review]
Would be great if `tests/manual/windowattributes` could cover this, so features like `no-frame` and `skip-taskbar` can be tested more easily. I hope `Qt::Dialog` works.

**Pull request merged**